### DISCUSSION TITLE
Focal Point should be settable per asset field, as well as for the asset itself.

### DISCUSSION THREAD

**Discussion by MattWilcox** [craftcms/cms#8452] [Ideas]
### Description

Though the new Focal Point feature is great it's implementation is brittle. The assumption seems to be that an image can only have one focal point and that it will be the same wherever it is used. However we've found it to be the case that a single image asset may have multiple focal points depending on the context in which it is used.

i.e., the focal point of an image sometimes depends on its use in context, and is not inherent to the image itself.

For example, a blog post discussing a work of art may want to use a particular image of a sunset painting in its header as a very wide crop of the top third of the image, but the same image in the blog listing page should focus on a boat in the bottom third, because that works better as a square crop thumbnail.

With the current implementation of Focal Points, two duplicate images would be required to achieve this behaviour - and because the Assets do not indicate where the focal point is, it is impossible to know which Asset to use where without dropping into the Image Editor. IT's not possible to know from a listing or selection interface.

I'd like to see the current behaviour retained as it's a sensible default - but with an addition of being able to set a focal point for the specific instance of the Asset in question. And the actual focal point to be visible in the overall Entry UI.

**Comment by brandonkelly** [comment]
Currently Craft isn’t set up to store additional data on a relation. So that would need to come first.

**Comment by MattWilcox** [comment]
We're also used to this "set a focal point per asset field" behaviour with the Craft 2 plugin [Focus Point](https://github.com/smcyr/Craft-FocusPoint/blob/master/README.md); which works as a new field type I believe - but beyond that I understand none of it.

**Comment by lboaretto** [comment]
I also have several clients with this use case.
Sometimes they need to show another point in the image.

It would be very good to have an Focal Point Override on Asset relation!

**Comment by gmhenderson** [comment]
Ditto. Right now focal points are kind of useless for me since I can't override them. What about adding a `ignoreFocalPoint` option to transforms?

```
{% set transform = {
    mode: 'crop',
    width: 250,
    position: 'top-left',
    ignoreFocalPoint: true,
} %}
```

and then in `craftcms/cms/src/services/AssetTransforms.php` around line 1303:

```
default:
    if ($asset->getHasFocalPoint() and ! $transform->ignoreFocalPoint) {
        $position = $asset->getFocalPoint();
    // ...
```

Of course there's more to it than that, but at least then focal points could be overridden on a per-transform basis.
